docs/usage/node-readiness.md

```diff
@@ -0,0 +1,32 @@
+# Readiness of Shoot Worker Nodes
+
+## Background
+
+When registering new `Nodes`, kubelet adds the `node.kubernetes.io/not-ready` taint to prevent scheduling workload pods to the `Node` until the `Ready` condition gets `True`.
+However, the kubelet does not consider the readiness of node-critical pods.
+Hence, the `Ready` condition might get `True` and the `node.kubernetes.io/not-ready` taint might get removed, for example, before the CNI daemon pod (e.g. `calico-node`) has successfully placed the CNI binaries on the machine.
+
+This problem has been discussed extensively in kubernetes, e.g., in [kubernetes/kubernetes#75890](https://github.com/kubernetes/kubernetes/issues/75890).
+However, several proposals have been rejected because the problem can be solved by using the `--register-with-taints` kubelet flag and dedicated controllers ([ref](https://github.com/kubernetes/enhancements/pull/1003#issuecomment-619087019)).
+
+## Implementation in Gardener
+
+Gardener makes sure that workload pods are only scheduled to `Nodes` where all node-critical components required for running workload pods are ready. 
+For this, Gardener follows the proposed solution by the Kubernetes community and registers new `Node` objects with the `node.gardener.cloud/critical-components-not-ready` taint (effect `NoSchedule`).
+gardener-resource-manager's [`node` controller](../concepts/resource-manager.md#node-controller) reacts on newly created `Node` objects that have this taint.
+The controller removes the taint once all node-critical pods are ready (determined by checking the Pods' `Ready` condition).
+
+The `node` controller considers all pods with the label `node.gardener.cloud/critical-component=true` as node-critical.
+If there are `DaemonSets` that contain the `node.gardener.cloud/critical-component=true` label in their pod template, the `node` controller waits for corresponding daemon pods to be scheduled and to get ready before removing the taint.
```

Comment:
Isn't it enough to just check such `DaemonSet`s? Why do you need to look at all `Pod`s in addition? Isn't a node-critical component always backed by a `DaemonSet` (otherwise, it can't be node-critical, can it)?

Reply:
I was thinking about whether it's possible to only check `DaemonSet`s and rely on their `status`es. The documentation states that the `Node` controller already waits for the corresponding daemon pods to be scheduled and ready. The "readiness" information can probably be retrieved from the `status`, but not per (specific) node, and that's why you check the pods as well, right?
Anyways, I think I need to go through the next commits, this will probably clarify this :)